workspace/2019 DeepSpace/src/main/java/frc/robot/Lifter.java

```diff
@@ -42,75 +42,92 @@ public void Lower() {
+    }
+    
     public void GoToBottom() {
         if (m_hight.atBottom()) {
             m_lifter.set(0);
             loc = Location.bottom;
-        } else { m_lifter.set(-kMotorPowerLevel); }
+        } else {
+            m_lifter.set(-kMotorPowerLevel);
+        }
     }
+    
     public void GoToTop() {
         if (m_hight.atTop()) {
             m_lifter.set(0);
             loc = Location.top;
-        } else { m_lifter.set(kMotorPowerLevel); }
+        } else {
+            m_lifter.set(kMotorPowerLevel);
+        }
     }
 
+    // All these go to level functions would be much easier in python.
     public void GoToFirstBallLevel() {
         if (m_hight.atFirstBallLevel()) {
             m_lifter.set(0);
             loc = Location.ballFirstLevel;
-        }else if (m_hight.atBottom()) {
-            m_lifter.set(kMotorPowerLevel); 
-        }else if (m_hight.atSecondBallLevel() || m_hight.atThirdBallLevel()){
-            m_lifter.set(-kMotorPowerLevel); }  
+        } else if (m_hight.getDistance() < m_hight.ballFirstLevelHight) {
+            m_lifter.set(kMotorPowerLevel);
+        } else if (m_hight.getDistance() > m_hight.ballFirstLevelHight) {
+            m_lifter.set(-kMotorPowerLevel);
+        }
     }
+    
     public void GoToSecondBallLevel() {
         if (m_hight.atSecondBallLevel()) {
             m_lifter.set(0);
             loc = Location.ballSecondLevel;
-        }else if (m_hight.atBottom() || m_hight.atFirstBallLevel()) {
-            m_lifter.set(kMotorPowerLevel); 
-        }else if (m_hight.atThirdBallLevel() || m_hight.atTop()){
-            m_lifter.set(-kMotorPowerLevel); }  
+        } else if (m_hight.getDistance() < m_hight.ballSecondLevelHight) {
+            m_lifter.set(kMotorPowerLevel);
+        } else if (m_hight.getDistance() > m_hight.ballSecondLevelHight) {
+            m_lifter.set(-kMotorPowerLevel);
+        }
     }
+    
     public void GoToThirdBallLevel() {
         if (m_hight.atThirdBallLevel()) {
             m_lifter.set(0);
             loc = Location.ballThirdLevel;
-        }else if (m_hight.atBottom() || m_hight.atFirstBallLevel() || m_hight.atSecondBallLevel()) {
-            m_lifter.set(kMotorPowerLevel); 
-        }else if (m_hight.atTop()){
-            m_lifter.set(-kMotorPowerLevel); }  
+        } else if (m_hight.getDistance() < m_hight.ballThirdLevelHight) {
+            m_lifter.set(kMotorPowerLevel);
+        } else if (m_hight.getDistance() > m_hight.ballThirdLevelHight) {
+            m_lifter.set(-kMotorPowerLevel);
+        }
     }
 
     public void GoToFirstHatchLevel() {
         if (m_hight.atFirstHatchLevel()) {
             m_lifter.set(0);
-            loc = Location.hatchFirstLevel;
-        }else if (m_hight.atBottom() ) {
-            m_lifter.set(kMotorPowerLevel); 
-        }else if (m_hight.atThirdHatchLevel() || m_hight.atSecondHatchLevel()){
-            m_lifter.set(-kMotorPowerLevel); }  
+            loc = Location.ballHatchLevel;
```

Comment:
`ballHatchLevel` -> `hatchFirstLevel`.  This code won't compile. You should always make sure it compiles before trying to check it in.

Reply:
Okay. These two files will be revised tomorrow (Tuesday), so that the code is easier to understand.